### PR TITLE
[#233] ubuntu base python image

### PR DESCRIPTION
**Pull request by alexander-semenets**
- base-python-image is based on ubuntu minimal instead of alpine
- temp packages cleanup added
- airflow image is based on base-python-image
- python3.6 is used as default python3 interpreter
- packages installation updated in according to new distribution

### CODE REVIEW

**Comment by alexander-semenets** [comment]
CI test this please

**Comment by alexander-semenets** [comment]
@akharlamov need approve again after rebase, tests passed btw

File: k8s/edi/Dockerfile [hunk 1]
```diff
@@ -15,9 +15,12 @@
 #
 FROM legion/base-python-image
 
-RUN apk add --update \
-    ca-certificates curl openrc docker \
-    nginx pcre pcre-dev supervisor
+RUN curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add - && \
+    python3.5 /usr/bin/add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
```
Comment (akharlamov) [inline k8s/edi/Dockerfile]: Python 3.5?

File: k8s/airflow/Dockerfile [hunk 2]
```diff
@@ -79,19 +47,20 @@ RUN rm -rf \
...
+RUN pip3 install -r /home/airflow/requirements.txt && \
+    pip3 install $pip_extra_index_params legion$pip_legion_version_string && \
+    pip3 install $pip_extra_index_params legion_airflow$pip_legion_version_string
 
 RUN chown -R airflow: ${AIRFLOW_HOME} && chmod -R 775 ${AIRFLOW_HOME} && \
     chmod a+x /home/airflow/init.sh
 
-COPY create_secrets.py /home/airflow/create_secrets.py
-
 EXPOSE 8080
```
Comment (akharlamov) [inline k8s/airflow/Dockerfile]: Do we need all these layers separately?

File: k8s/edi/Dockerfile (hunk 1 above)
Comment (alexander-semenets) [reply k8s/edi/Dockerfile]: apt-key doesn't work with python 3.6 so have to explicitly specify 3.5 interpreter

File: base-python-image/Dockerfile [hunk 3]
```diff
@@ -13,20 +13,26 @@
...
-    tiff-dev jpeg-dev zlib-dev tcl-dev tk-dev python2-tkinter \
-    python3-tkinter-tests freetype-dev lcms2-dev libwebp harfbuzz \
-    libstdc++ musl-dev linux-headers build-base pcre pcre-dev\
-    openblas-dev freetype-dev pkgconfig \
-    && ln -s locale.h /usr/include/xlocale.h
+ENV DEBIAN_FRONTEND=noninteractive
 
-ADD requirements.txt /root/requirements.txt
+# Add Python3.6 repository
+RUN apt-get update && apt-get install -y software-properties-common && \
```
[the remark this answers is not in the document]
Comment (alexander-semenets) [reply base-python-image/Dockerfile]: we have to add repositories and apt tools before proceeding, also python repo and basic ubuntu tools supposed to be changed much rarely then dependencies packages and libs so makes sense to have them in separate layers for better layers reuse

**Comment by alexander-semenets** [comment]
@MaksymBilenko could you please take a brief look and share your opinion regarding the proposed Dockerfiles structure

**Comment by alexander-semenets** [comment]
test this please

**Comment by dsuslov** [comment]
👍

**Pull request merged**